assertsprocessor/priority_queue.go

```diff
@@ -0,0 +1,100 @@
+package assertsprocessor
+
+import (
+	"container/heap"
+	"context"
+	"go.opentelemetry.io/collector/pdata/ptrace"
+	"sync"
+)
+
+// An Item is something we manage in a latency queue.
+type Item struct {
+	trace   *ptrace.Traces // The value of the item; arbitrary.
+	ctx     *context.Context
+	latency float64 // The latency of the item in the queue.
+	// The index is needed by update and is maintained by the heap.Interface methods.
+	index int // The index of the item in the heap.
+}
+
+// A PriorityQueue implements heap.Interface and holds Items.
+type PriorityQueue []*Item
+
+type TraceQueue struct {
+	priorityQueue PriorityQueue
+	maxSize       int
+	mutex         sync.Mutex
+}
+
+func NewTraceQueue(maxSize int) *TraceQueue {
+	traceQueue := TraceQueue{
+		priorityQueue: make(PriorityQueue, 0),
+		maxSize:       maxSize,
+	}
+	return &traceQueue
+}
+
+func (qw *TraceQueue) push(item *Item) {
```

Comment:
Why is the argument named `qw`?

Reply:
> Why is the argument named `qw`?

The `TraceQueue` wraps around the priority queue, so I think I had `qw` in mind. `tq` would be better choice.